src/twisted/logger/test/test_util.py

```diff
@@ -130,4 +109,31 @@ def testObserver(e: LogEvent) -> None:
 
         root = LogPublisher(o1, p1, o5, oTest)
         root.name = "root/"  # type: ignore[attr-defined]
+        expectedTraceTemplate = (
+            "{root} ({root.name})\n"
+            "  -> {o1} ({o1.name})\n"
+            "  -> {p1} ({p1.name})\n"
+            "    -> {o2} ({o2.name})\n"
+            "    -> {o3} ({o3.name})\n"
+            "    -> {p2} ({p2.name})\n"
+            "      -> {o4} ({o4.name})\n"
+            "  -> {o5} ({o5.name})\n"
+            "  -> {oTest}\n"
+        )
+
+        # Mypy rightly complains that we're pulling some Shenanigans with all
+        # these 'name' attributes above (LogPublisher does not have any such
+        # attribute, neither does FunctionType) so we split up the 'format'
+        # call so it can't see the attributes being formatted.
+        expectedTrace = expectedTraceTemplate.format(
+            root=root,
+            o1=o1,
+            o2=o2,
+            o3=o3,
+            o4=o4,
+            o5=o5,
+            p1=p1,
+            p2=p2,
+            oTest=oTest,
+        )
```

Comment:
Are said "shenanigans" a thing that would prevent us from using f-strings here?

Reply:
Yeah. The shenanigan is that we want to grab this "name" attribute that is not properly predeclared, and mypy (rightly) complains if it has visibility into the code that's being executed in the string.

Reply:
(fstrings would obviously improve the visibility even more, which would make it worse.)